packages/components/src/core/ComplexFilter/index.tsx

```diff
@@ -176,25 +176,17 @@ const ComplexFilter = <
       if (isTriggerChangeOnOptionClick) {
         setPendingValue(newValue);
 
-        return setValueAndCallOnChange(event, newValue);
+        return setValue(newValue);
       }
 
       return setPendingValue(newValue);
     }
 
-    setValueAndCallOnChange(event, newValue);
+    setValue(newValue);
 
     if (!multiple) setOpen(false);
   }
 
-  function setValueAndCallOnChange(
```

Comment:
We are calling the onChange in a useEffect at line 76, so there is no need for this function anymore!